src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Tests/Controllers/PublicationsControllerTests.cs

```diff
@@ -82,24 +73,23 @@ public ListPublicationsTests(TestApplicationFactory testApp) : base(testApp)
                         .SetDefault(p => p.Theme)
                         .Set(p => p.Published, p => p.Date.Past())
                         .Set(p => p.Type, ReleaseType.OfficialStatistics)
-                        .Set(p => p.Id, publicationId))
-                    .Generate();
-
-                publications.Add(publication);
-            }
+                        .Set(p => p.Id, id))
+                    .Generate()
+                )
+                .ToList();
 
             var contentApiClient = new Mock<IContentApiClient>();
             contentApiClient
                 .Setup(c => c.ListPublications(
                     page,
                     pageSize,
                     null,
-                    It.Is<IEnumerable<Guid>>(ids => 
+                    It.Is<IEnumerable<Guid>>(ids =>
                         Enumerable.SequenceEqual(
                             ids.Order(),
                             publishedDataSetPublicationIds.Order()))))
```

Comment:
I think I need to order them because we can't guarantee the order in which EF is going to save the entities to the DB. So it would be a flakey test if we didn't order them first; hence the need for the `It.Is`

Reply:
If you try the test with a few more publications, say set `numberOfPublishedDataSets = 9` and run the test, you will probably see it fail (due to the higher probability of the order being different so the mock doesn't work).

But it will work every time if you order them

Reply:
Fair enough, makes sense then!